src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -318,4 +318,66 @@ object messages {
         |$code2
         |""".stripMargin
   }
+
+  case class TopLevelImplicitClass(cdef: untpd.TypeDef)(implicit ctx: Context)
+    extends Message(10) {
+    val kind = "Syntax"
+
+    val msg = hl"""|An ${"implicit class"} may not be top-level"""
+
+    val explanation =
+    hl"""|implicit class are not allowed to be defined at the top-level.
+         |
+         |Declare ${cdef.name} inside of an ${"object"} or ${"package object"}
+         |then import the members of the object at the use site if needed, for example:
+         |
+         |package object Implicits {
+         |  implicit class ${cdef.name}...
```

Comment:
I would have liked to add the args here but did not see how to grab them.  Can I use scala.meta for this?